### ISSUE TITLE
1.0.5 - Reoccurring Events Seem Buggy

### ISSUE THREAD

**Issue by MNOCFan** [libasys/calendarplus#118]
Issue 1
1) Add a reoccurring event over M-F. 
2) Delete Wednesdays event. 
Result: Thursday and Fridays events back up a day to Wednesday and Friday is ultimately removed.

Issue 2
1) Add a reoccurring event over M-F when running over the end of the month to the beginning of the next. Example: Aug 31st-Sept 4th.
2) At first, not all the events show on the calendar until the calendar is refreshed.
3) Deleting an event somewhere in the middle causes more weird behavior.  
##

**Comment by pravussum** [comment]
I can add something to issue 2:
It seems that in month view the recurring events are only recalculated up to the end of the month. If there are some days of the next month visible and one recurrence of the event in that next month is visible as well, it only gets redrawn on refresh). I hope I made myself clear enough :-/

**Comment by MNOCFan** [comment]
@pravussum While I can almost agree with you, in one of my tests I did see the event get created on the 31st and the 1st, but not on the 2nd, 3rd, or 4th date. Still, something buggy about it....

**Comment by libasys** [comment]
please update your app to current master 1.0.5, this should be fixed! Added very important js fix fot timeline, if not no events loaded!

**Comment by MNOCFan** [comment]
@libasys I have updated and this issue is still occurring.

**Comment by libasys** [comment]
ok i can't reproduce that behaviour! I have no problem with creation of a recurring rule of an event from monday till friday!

**Comment by MNOCFan** [comment]
@libasys There doesn't seem to be an issue with creating an event reoccurring from M-F, however, after creating that event, try deleting one event within that series. Try deleting Wednesday or something in the middle of the events week. Pay attention to the events after the one you delete. It doesn't behave as intended.

**Comment by libasys** [comment]
I have tested this with deletion, i have no Problems! Happens that with allday events?

**Comment by MNOCFan** [comment]
@libasys here are the steps I am doing to reproduce:
1) Create an event on Monday and mark it to reoccur daily until Friday
2) Delete the Wednesday within that reoccurring event
3) In my environment, after deleting Wednesday, Thursdays event moves to Wednesday. 
4) Now my week looks like this - Event Monday, Tuesday, Wednesday, and Friday.

**Comment by ev0lister** [comment]
Cannot reproduce the issue you are describing. When I delete a event from a series it is just working fine. Using current version 1.0.7. @MNOCFan have you tried a fresh calendar to see whether this is maybe related to other entries in your current calendar?

![2015-08-27 14_12_16-calendar](https://cloud.githubusercontent.com/assets/6650205/9520247/d607e6c4-4cc5-11e5-8f2a-51d219e968cf.png)

**Comment by pravussum** [comment]
I can kindof reproduce this... 
- week view from August 24th to 30th
- Created an event from monday, recurring daily, ending after 28th (Friday)
- problem 1: event was created up to Thursday
- edited event, to end after 29th (Saturday) so its recurring up to Friday
- clicked on Wednesday event, deleted event (exception for 26th)
- but exception for Tuesday (25th is added), making the events effectively "moving" like @MNOCFan described

basicly: 
- exception is added for wrong day
- exception end calculation is off by one day

**Comment by pravussum** [comment]
Same use case works in the week after. Could it be, that it depends on whether the current day is within the timespan?
@ev0lister: can you try that in the current week please?

**Comment by libasys** [comment]
have you tested the advanced repeating rules like -> customize -> weekly -> choose mo,tue,thu,fr? without wednesday?

**Comment by pravussum** [comment]
Just tested (in the current week), that works.

**Comment by ev0lister** [comment]
Have tested it in current week with non-advanced rules (e.g. just selected DAILY) and this is also working for me.

**Comment by libasys** [comment]
Please retest current master of calendar+! Should be fixed!

**Comment by MNOCFan** [comment]
I have tested this with the latest version (1.0.7) with an existing calendar and a brand new calendar and it still behaves buggy. Here are my steps to recreate. 
- Viewing calendar in either month mode or weekly mode (I have tested both)
- Create an event on Monday
- Mark this event reoccurring daily (end by date) until the Friday of the same week
- Now, you have a reoccurring event M-F for one week
- Next, delete the even for Wednesday of that week
- Finally, in my environment, after deleting the event from Wednesday, the event from Thursday disappears and moves to Wednesday.

I certainly hope I am not the only one experiencing this issue....

**Comment by libasys** [comment]
Have you tested this Master here on github or downloaden from AppStore? The fix is only here on Master !

**Comment by MNOCFan** [comment]
@libasys I only did the one-click update. I will try downloading from Github.

**Comment by MNOCFan** [comment]
@libasys I have tested with the Github version with the same results. I notice there are "Exceptions" listed within the series. They are not the correct dates that I deleted rather they are the adjacent. I will try to get some snapshots to post of what I am seeing.

**Comment by MNOCFan** [comment]
I added a new event from M-F
![addevent](https://cloud.githubusercontent.com/assets/7968381/9530032/2f0f4382-4cc3-11e5-8a9b-69124bff4aa5.JPG)
![event](https://cloud.githubusercontent.com/assets/7968381/9530073/5a398612-4cc3-11e5-834d-e3a60c7c0a5a.JPG)
Deleting the Wednesday event.
![deleteevent](https://cloud.githubusercontent.com/assets/7968381/9530102/7ca51d1a-4cc3-11e5-9082-d6a1fcbd542e.JPG)
After deleting Wednesday, Thursday disappears. 
![afterdelete](https://cloud.githubusercontent.com/assets/7968381/9530134/9cae449c-4cc3-11e5-85ce-6fc9bc79c33e.JPG)
Additionally, the "exception" date is wrong.
![theexceptionwrong](https://cloud.githubusercontent.com/assets/7968381/9530169/d7e9a0ba-4cc3-11e5-98a5-ae79e25e0277.JPG)

**Comment by libasys** [comment]
I think this a timezone issue, what timezone do you have selected?

**Comment by MNOCFan** [comment]
Chicago

**Comment by libasys** [comment]
I can't reproduce this issue, i changed tz to Chicago and changed dateformat but all Works well! Can you reproduce this on the Demo http://demo81.libasyscloud.de ?

**Comment by MNOCFan** [comment]
@libasys Yes, on your demo link I just added 9/7-9/11 and deleted 9/9 and 9/10 deleted instead. 
Translation was tough, but I made through it! :)
![demosite](https://cloud.githubusercontent.com/assets/7968381/9531148/ea1daca8-4cc9-11e5-86e8-80eda20147df.JPG)

**Comment by MNOCFan** [comment]
@libasys I uploaded a video onto your demo site under files --> videos called Cal+_Issue2.mp4

Let me know if you can see it ok or not.

**Comment by pravussum** [comment]
Did a pull from master and just tested with exact dates from @MNOCFan in week view and Chicago time zone - worked....
Tried in the current week - worked as well.
Calculation of the recurrence end date also worked correctly (in both Chicago and Berlin timezone).
So now it gets interesting...

**Comment by libasys** [comment]
Ok i can't reproduce testing your example! What Browser and operating System do you use?

**Comment by pravussum** [comment]
@libasys maybe you can set your OS time to Chicago...

**Comment by MNOCFan** [comment]
I am on a Windows 8.1 Pro machine and am using latest version of Firefox.

**Comment by MNOCFan** [comment]
I just tested this on my Ubuntu machine with Firefox and got the same results.

**Comment by MNOCFan** [comment]
@libasys @pravussum UPDATE: So I am not sure if you guys were testing the reoccurring events by the full day or for a certain time or not, but I have discovered that the "Full Day" reoccurring event is what is broken. If I create an even for a certain time-frame and delete one of the events, it works as expected.

**Comment by pravussum** [comment]
Retested full day events from 7th of Sep to 11th of Sep with timezone set to Chicago. Deleted the Wednesday event and Wednesday event was removed correctly. 
So everything working in my setup (on current master).
